verifier/src/evidence.rs

```diff
@@ -188,168 +374,6 @@ mod test {
         Collateral::try_from(&sgx_collateral).expect("Failed to parse collateral")
     }
 
-    fn tcb_signing_key() -> VerifyingKey {
-        let pem = include_str!("../data/tests/tcb_signer.pem");
-        let certificate = Certificate::from_pem(pem).expect("failed to parse PEM");
-        let key = VerifyingKey::from_sec1_bytes(
-            certificate
-                .tbs_certificate
-                .subject_public_key_info
-                .subject_public_key
-                .as_bytes()
-                .expect("Failed to parse public key"),
-        )
-        .expect("Failed to decode public key");
-
-        key
-    }
-
-    fn quote_signing_key<T: AsRef<[u8]>>(quote: &Quote3<T>) -> VerifyingKey {
-        let signature_data = quote.signature_data();
-        let certification_data = signature_data.certification_data();
-        let CertificationData::PckCertificateChain(chain) = certification_data else {
-            panic!("Unexpected certification data type");
-        };
-        let leaf_pem = chain.into_iter().next().expect("No leaf certificate");
-        let certificate =
-            Certificate::from_pem(leaf_pem).expect("Failed to parse leaf certificate");
-
-        let key = VerifyingKey::from_sec1_bytes(
-            certificate
-                .tbs_certificate
-                .subject_public_key_info
-                .subject_public_key
-                .as_bytes()
-                .expect("Failed to parse public key"),
-        )
-        .expect("Failed to decode public key");
-
-        key
-    }
-
-    type VerifierType<T> = And<
-        Quote3Verifier<T>,
-        And<
```

Comment:
I think the `EvidenceVerifier` IFF it's the way forward removes the main need to try and build up a verifier of all the necessary pieces operating on the `Evidence` type.

I'm still not sure how/where we'll want to handle the Debug flag, https://github.com/mobilecoinfoundation/mobilecoin/issues/3376